### PR TITLE
Expand readme, link to Lux.jl

### PR DESCRIPTION
**Pull request by mcabbott**

### CODE REVIEW

**Review by ToucheSir — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -20,14 +20,37 @@
...
+state, and the model with its trainable parameters adjusted:
+
+```julia
+state = Optimisers.setup(Optimisers.ADAM(), model)  # just once
+
+state, model = Optimisers.update(state, model, grad)  # at every step
+```
+
+For models with deeply nested layers containing the parameters (like [Flux.jl](https://github.com/FluxML/Flux.jl) models),
+this state is an similar tree.
```
Comment (CarloLucibello) [inline README.md:45]:
```suggestion
this state is similar to a tree.
```
?

File: README.md [hunk 2]
```diff
@@ -20,14 +20,37 @@
...
+and returns this along with a function to re-build a similar model:
+
+```julia
+vector, re = Optimisers.destructure(model)
+
+model2 = re(2 .* vector)
+```
+
+[The documentation](https://fluxml.ai/Optimisers.jl/dev/) explains in more detail,
+describes all the optimsation rules, and shows how to define new ones.
```
Comment (CarloLucibello) [inline README.md:56]:
```suggestion
describes all the optimisation rules, and shows how to define new ones.
```

File: README.md (hunk 1 above)
Comment (darsnack) [inline README.md:45]:
```suggestion
this state is a tree similar to the model structure.
```

File: README.md [hunk 3]
```diff
@@ -20,14 +20,37 @@
...
+The function `destructure` collects all the trainable parameters into one vector,
+and returns this along with a function to re-build a similar model:
+
+```julia
+vector, re = Optimisers.destructure(model)
+
+model2 = re(2 .* vector)
+```
+
+[The documentation](https://fluxml.ai/Optimisers.jl/dev/) explains in more detail,
```
Comment (darsnack) [inline README.md:55]:
```suggestion
[The documentation](https://fluxml.ai/Optimisers.jl/dev/) explains this in more detail,
```

File: README.md (hunk 1 above)
Comment (mcabbott) [reply README.md:45]: I guess I was aiming for:
```suggestion
this state is a similarly nested tree.
```

**Pull request merged**